src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,52 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F, S> {
+    private F firstValue;
+    private S secondValue;
 
+    private Pair(F firstValue, S secondValue) {
+        this.firstValue = firstValue;
+        this.secondValue = secondValue;
+    }
+
+    public F getFirst() {
+        return firstValue;
+    }
+
+    public S getSecond() {
+        return secondValue;
+    }
+
+    static <F,S> Pair of(F first, S second) {
+        return new Pair(first,second);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 1;
+        result = getFirst().equals(null) ? result : 31 * getFirst().hashCode();
+        result = getSecond().equals(null) ? result : 31 * getSecond().hashCode();
+
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Pair)) {
```

Comment:
what is more correct : getClass() or instanceof  and why?

Reply:
Made getClass, because we test  the actual type of the argument,  not a reference.